src/manager/index.js

```diff
@@ -0,0 +1,160 @@
+import picoSignals from 'pico-signals';
+import createPGroup from './util/p-group';
+
+const findLocaleById = (locales, localeId) => {
+    const locale = locales.find(({ id }) => id === localeId);
+
+    if (!locale) {
+        throw new Error(`Unknown locale id: ${localeId}`);
+    }
+
+    return locale;
+};
+
+const match = (locales, policies, ctx = {}) => {
+    const localeId = policies.reduce((localeId, policy) => localeId ?? policy.match(locales, ctx), null);
+
+    if (!localeId) {
+        throw new Error('None of the policies matched a locale.. did you forgot to include the default policy?');
+    }
+
+    return findLocaleById(locales, localeId);
+};
+
+const save = async (policies, locale) => {
+    const policy = policies.find((policy) => policy.save);
+
+    if (policy) {
+        await policy.save(locale);
+    }
+};
+
+const act = (policies, locale) => {
```

Comment:
What's exactly the purpose and role of `act` in the manager itself and in each policy?

Reply:
While it’s not being used yet by any policy, it will be used by #2. In this policy, the act method would listen to route change events (popstate), in order to fix the the route (with replaceState) to match the current locale. Please note that the act method may return a cleanup function, which would unsubscribe to popstate events.

Reply:
Gotcha! 👌